Mathlib/RingTheory/TwoSidedIdeal/Basic.lean

```diff
@@ -156,17 +156,16 @@ def mk' (carrier : Set R)
         rw [show a + c - (b + d) = (a - b) + (c - d) by abel]
         exact add_mem h1 h2 }
 
-lemma mem_mk' (carrier : Set R)
-    (zero_mem : 0 ∈ carrier)
-    (add_mem : ∀ {x y}, x ∈ carrier → y ∈ carrier → x + y ∈ carrier)
-    (neg_mem : ∀ {x}, x ∈ carrier → -x ∈ carrier)
-    (mul_mem_left : ∀ {x y}, y ∈ carrier → x * y ∈ carrier)
-    (mul_mem_right : ∀ {x y}, x ∈ carrier → x * y ∈ carrier)
-    (x : R) :
-    x ∈ mk' carrier zero_mem add_mem neg_mem mul_mem_left mul_mem_right ↔ x ∈ carrier := by
+lemma mem_mk' (carrier : Set R) (h₁ h₂ h₃ h₄ h₅) (x : R) :
```

Comment:
Why the change to the variable names?

Reply:
The change to names is inconsequential (though I do think it's more readable not to have so many letters in the statement), but the change to their _types_ (namely, omitting the types and letting Lean infer them) is what let me simplify the proof of `mem_ker`. The difference in types is just whether something is eta-expanded, but the new version allows Lean to infer them and the previous one didn't.

Reply:
I guess the eta expanding could be avoided by writing `@one_mem` etc instead, but your version is certainly easier to read. The fact that eta-expansions are performed (presumably thanks to implicit lambdas) is quite surprising to me.

Reply:
I would prefer to keep the names, please. Reasoning: `mk'` should actually be the default method for defining a `TwoSidedIdeal`, and users will probably need to apply `mem_mk'` soon thereafter. It would be a shame if they had to read the actual types of the hypotheses in order to do so, as opposed to just seeing the names and knowing intuitively what they are.

Reply:
I readded the names, though I don't think anybody should be providing these arguments manually: they should be filled in by unification.